adapters/repos/db/lsmkv/store.go

```diff
@@ -103,6 +108,39 @@ func (s *Store) bucketDir(bucketName string) string {
 	return path.Join(s.dir, bucketName)
 }
 
+// lockBucket it locks a specific bucket by it's name
+// to hold ant concurrent access to that specific bucket
+//
+//	don't forget calling unlockBucket() after locking it.
+func (s *Store) lockBucket(bucketName string) {
+	var bucketLock *sync.Mutex
+	var ok bool
+
+	s.bucketLevelLock.Lock()
+	defer s.bucketLevelLock.Unlock()
+
+	bucketLock, ok = s.bucketsLocks[bucketName]
+	if !ok {
+		bucketLock = &sync.Mutex{}
+		s.bucketsLocks[bucketName] = bucketLock
+	}
+
+	bucketLock.Lock()
+}
+
+// unlockBucket it unlocks a specific bucket by it's name
+func (s *Store) unlockBucket(bucketName string) {
+	s.bucketLevelLock.Lock()
+	defer s.bucketLevelLock.Unlock()
+
+	bucketLock, ok := s.bucketsLocks[bucketName]
+	if !ok {
+		return
```

Comment:
this may be hiding a bug, it may be better to panic or at least raise an error when trying to unlock if it's not already locked. so to be consistent with the sync.mutex behaviour

Reply:
i didn't want to error on it , instead I made an error log though 🤔

Reply:
hm, I'd think the log is something too soft... that situation should not happen unless there is a real bug in the code...